C/services/south/ingest.cpp

```diff
@@ -522,15 +527,35 @@ void Ingest::processQueue()
                                         const std::vector<Datapoint *> dpVec = reading->getReadingData();
 					std::string temp;
 
+					std::set<std::string> tempSet;
+					// first sort the individual datapoints 
+					// e.g. dp2, dp3, dp1 push them in a set,to make them 
+					// dp1,dp2,dp3
                                         for ( auto dp : dpVec)
                                         {
+						temp.clear();
                                                 temp.append("\"");
                                                 temp.append(dp->getName());
                                                 temp.append("\"");
-                                                assetDatapointMap[assetName].insert(temp);
+						tempSet.insert(temp);
                                         }
 
-                                       	assetDatapointMap[assetName].insert(temp);
+					temp.clear();
+
+					// make a string from sorted datapoints in a reading 
+					int i = 0;
+					for (auto setItr: tempSet)
+					{
+						if ( i> 0) temp.append(",");
+						temp.append(setItr);
+						++i;
+					}
+
+					// Push them in a set so as to avoid duplication of datapoints
+					// a reading of d1, d2, d3 and another d2,d3,d1 , second will be discarded
+
+					assetDatapointMap[assetName].setDp.insert(temp);
+                                        assetDatapointMap[assetName].count = i;
```

Comment:
Do we always insert the dpset into map? Or only when the count has increased from previous value? If we insert always, then finally before updating cache, do we find the dpset with the maximum count for each asset seen?

Reply:
whether to insert in db or not is checked in find function in the storageassettracker